### PR TITLE
Implement Tunnistamo OpenId Connect authentication

### PR DESCRIPTION
**Pull request by Jogha**
This PR will implement OpenID Connect authentication flow, with some added Tunnistamo API token stuff to top that off.

Redux-oidc package is the base for this implementation, and is doing a lot of heavy lifting here. Tunnistamo uses separated API tokens, so fetching those is implemented together with redux-oidc.

### CODE REVIEW

File: src/reducers/api.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+};
+
+const fetchApiToken = (state) => ({
+  ...state,
+  isFetching: true
+});
+
+const receiveApiToken = (state, {payload}) => {
+  if (payload) {
+    return {isFetching: false, jwt: payload['https://auth.turku.fi/kerrokantasi']};
```
[the remark this answers is not in the document]
Comment (Jogha) [reply src/reducers/api.js]: Yes, somehow this was left out of configurations... I will add it there.

File: src/views/Hearing/HearingContainer.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ const HearingEditor = lazy(() => import(/* webpackChunkName: "editor" */'../../c
 
 export class HearingContainerComponent extends React.Component {
   componentWillMount() {
+    console.error("HEARTING CONTAINER");
```
Comment (tituomin) [inline src/views/Hearing/HearingContainer.js]: Should probably be removed.

**Comment by tituomin** [comment]
Great work, a few small comments above.

File: src/config.js [hunk 3]
```diff
@@ -4,8 +4,13 @@ const config = {
   pluginMap,
   languages: ['fi', 'sv', 'en'],
   activeLanguage: 'fi',
+  publicUrl: (typeof window !== 'undefined' ? window.PUBLIC_URL : null) || 'http://localhost:8086/',
   apiBaseUrl: (typeof window !== 'undefined' ? window.API_BASE_URL : null) || 'http://localhost:8000/',
   uiConfig: typeof window !== 'undefined' ? window.UI_CONFIG : null,
+  openIdClientId: typeof window !== 'undefined' ? window.OPENID_CLIENT_ID : null,
```
Comment (Rikuoja) [inline src/config.js]: Here we expose the client ID to the client for anybody to see. Is the client id public data??

Comment (vikoivun) [reply src/config.js]: It is indeed. It is included in the “authorize” call made in users browser.

Comment (Rikuoja) [reply src/config.js]: Ah yeah. And the client id may indeed be a legible string (like in https://github.com/City-of-Helsinki/kukkuu-ui/blob/develop/.env#L5), as it does not need to be secret.

**Comment by vikoivun** [comment]
I can pick up the documentation side.

**Pull request merged**